velox/dwio/parquet/tests/ParquetTpchTest.cpp

```diff
@@ -0,0 +1,352 @@
+             DOUBLE(),
+             VARCHAR(),
+             VARCHAR(),
+             DATE(),
+             DATE(),
+             DATE(),
+             VARCHAR(),
+             VARCHAR(),
+             VARCHAR()});
+  }
+
+  void TearDown() override {
+    parquet::unregisterParquetReaderFactory();
+    HiveConnectorTestBase::TearDown();
+  }
+
+  int64_t date(std::string_view stringDate) const {
+    Date date;
+    parseTo(stringDate, date);
+    return date.days();
+  }
+
+  // Split file at a given path 'filePath' into 'numSplits' splits.
+  std::vector<std::shared_ptr<connector::hive::HiveConnectorSplit>> makeSplits(
+      const std::string& filePath,
+      int64_t numSplits) const {
+    const int fileSize = fs::file_size(filePath);
+    // Take the upper bound.
+    const int splitSize = std::ceil(fileSize / numSplits);
+    std::vector<std::shared_ptr<connector::hive::HiveConnectorSplit>> splits;
+
+    // Add all the splits.
+    for (int i = 0; i < numSplits; i++) {
+      auto split = makeHiveConnectorSplit(filePath, i * splitSize, splitSize);
+      split->fileFormat = dwio::common::FileFormat::PARQUET;
+      splits.push_back(std::move(split));
+    }
+    return splits;
+  }
+
+  void addSplitsToTask(
+      exec::Task* task,
+      bool& noMoreSplits,
+      const std::vector<std::string>& filePaths,
+      int sourcePlanNodeId = 0,
+      int64_t numSplits = 10) const {
+    if (!noMoreSplits) {
+      for (const auto& filePath : filePaths) {
+        auto const& splits = makeSplits(filePath, numSplits);
+        for (const auto& split : splits) {
+          task->addSplit(std::to_string(sourcePlanNodeId), exec::Split(split));
+        }
+      }
+      task->noMoreSplits(std::to_string(sourcePlanNodeId));
+      noMoreSplits = true;
+    }
+  }
+
+  // Write the DuckDB Lineitem TPC-H table to a Parquet file and return the file
+  // location.
+  std::string writeLineitemTableToParquet() const {
+    constexpr std::string_view tableName("lineitem");
+    constexpr int kRowGroupSize = 10'000;
+    const auto& filePath =
+        fmt::format("{}/{}.parquet", tempDirectory_->path, tableName);
+    // Convert decimal columns to double.
+    const auto& query = fmt::format(
+        "COPY (SELECT l_orderkey as orderkey, l_partkey as partkey, l_suppkey as suppkey, l_linenumber as linenumber, "
+        "l_quantity::DOUBLE as quantity, l_extendedprice::DOUBLE as extendedprice, l_discount::DOUBLE as discount, "
+        "l_tax::DOUBLE as tax, l_returnflag as returnflag, l_linestatus as linestatus, "
+        "l_shipdate as shipdate, l_receiptdate as receiptdate, "
+        "l_shipinstruct as shipinstruct, l_shipmode as shipmode, l_comment as comment "
+        "FROM {}) TO '{}' (FORMAT 'parquet', ROW_GROUP_SIZE {})",
+        tableName,
+        filePath,
+        kRowGroupSize);
+    duckDb_->execute(query);
+    return filePath;
+  }
+
+  RowTypePtr getLineitemColumns(
+      std::vector<std::string>&& outputColumnNames) const {
```

Comment:
I'd drop && since there is no need to force the caller to std::move the names. I'd also shorten the parameter name to "names" for readability.

```
RowTypePtr getLineitemColumns(std::vector<std::string> names) const
```